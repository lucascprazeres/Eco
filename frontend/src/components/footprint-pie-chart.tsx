import { useFootprint } from '@eco/providers/footprint-provider'
import { toPercentage } from '@eco/utils/percentage'
import { PieChart } from '@mui/x-charts'
import { useMemo } from 'react'

export function FootprintPieChart() {
  const { footprint } = useFootprint()

  const pieChartData = useMemo(() => {
    if (!footprint) return []

    return [
      {
        label: 'Electricity Usage (%)',
        value: toPercentage(footprint.electricity.value, footprint.total.value),
      },
      {
        label: 'Transportation fuel (%)',
        value: toPercentage(
          footprint.transportation.value,
          footprint.total.value,
        ),
      },
      {
        label: 'Air travel (%)',
        value: toPercentage(footprint.airTravel.value, footprint.total.value),
      },
    ]
  }, [footprint])

  return (
    <PieChart
      series={[
        {
          data: pieChartData,
          cx: 0,
          highlightScope: { faded: 'global', highlighted: 'item' },
          faded: {
            innerRadius: 20,
            additionalRadius: -20,
            color: 'gray',
          },
        },
      ]}
      height={350}
      width={300}
      colors={['#34CB79', '#322153', '#8257E5']}
      slotProps={{ legend: { padding: -40 } }}
    />
  )
}
