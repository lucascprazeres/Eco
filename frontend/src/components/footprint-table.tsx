import { useFootprint } from '@eco/providers/footprint-provider'
import {
  Paper,
  styled,
  Table,
  TableBody,
  TableCell,
  tableCellClasses,
  TableContainer,
  TableHead,
  TableRow,
} from '@mui/material'
import { useMemo } from 'react'

const StyledTableCell = styled(TableCell)(({ theme }) => ({
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.background.default,
    fontSize: 16,
    fontWeight: '700',
  },
  [`&.${tableCellClasses.body}`]: {
    fontSize: 15,
  },
}))

export function FootprintTable() {
  const { footprint } = useFootprint()

  const footprintBreakdownRows = useMemo(() => {
    if (!footprint) return []

    return [
      {
        name: 'Electricity Usage  (Kwh/year)',
        emission: `${footprint?.electricity?.value} ${footprint?.electricity?.unit}`,
      },
      {
        name: 'Transportation fuel  (Gallons/year)',
        emission: `${footprint?.transportation?.value} ${footprint?.transportation?.unit}`,
      },
      {
        name: 'Air travel  (travels/year)',
        emission: `${footprint?.airTravel?.value} ${footprint?.airTravel?.unit}`,
      },
      {
        name: 'Total',
        emission: `${footprint?.total?.value} ${footprint?.total?.unit}`,
      },
    ]
  }, [footprint])

  return (
    <TableContainer variant="outlined" component={Paper} sx={{ width: 600 }}>
      <Table>
        <TableHead>
          <TableRow>
            <StyledTableCell>Name</StyledTableCell>
            <StyledTableCell>Emission</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {footprintBreakdownRows.map((row) => (
            <TableRow
              key={row.name}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <StyledTableCell>{row.name}</StyledTableCell>
              <StyledTableCell sx={{ fontWeight: 'bold' }}>
                {row.emission}
              </StyledTableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}
