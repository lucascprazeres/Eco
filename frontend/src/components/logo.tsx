import { Box, Typography } from '@mui/material'
import Image from 'next/image'

import logoImg from '@eco/assets/logo.svg'

export function Logo() {
  return (
    <Box display="flex" alignItems="center" gap={1}>
      <Typography
        variant="h4"
        color="primary.dark"
        fontFamily="ubuntu"
        fontWeight="700"
      >
        Eco
      </Typography>
      <Image src={logoImg} alt="" height={40} width={40} />
    </Box>
  )
}
