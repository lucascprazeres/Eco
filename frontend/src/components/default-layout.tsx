import { Box } from '@mui/material'
import { ReactElement } from 'react'
import { Logo } from './logo'

interface DefaultLayoutProps {
  children: ReactElement | ReactElement[]
}

export function DefaultLayout({ children }: DefaultLayoutProps) {
  return (
    <Box height="100vh" width="100vw">
      <Box component="header" width="100%" padding={2} height={72}>
        <Logo />
      </Box>

      <Box
        height="calc(100% - 72px)"
        component="main"
        display="flex"
        flexDirection="column"
        alignItems="center"
        justifyContent="center"
      >
        {children}
      </Box>
    </Box>
  )
}
