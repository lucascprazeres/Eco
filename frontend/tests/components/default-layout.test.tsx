import { describe, it, expect, afterEach, vi } from 'vitest'
import { render, screen, cleanup } from '@testing-library/react'
import { DefaultLayout } from '@eco/components/default-layout'

describe('DefaultLayout', () => {
  afterEach(() => {
    cleanup()
    vi.clearAllMocks()
  })

  it('should be able to render correcly', () => {
    // arrange/ act
    const result = render(
      <DefaultLayout>
        <div>Test Content</div>
      </DefaultLayout>,
    )

    // assert
    const child = screen.getByText('Test Content')

    expect(child).toBeInTheDocument()
    expect(result).toMatchSnapshot()
  })
})
