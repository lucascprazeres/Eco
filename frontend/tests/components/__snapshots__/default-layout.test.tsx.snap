// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultLayout > should be able to render correcly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-21eb5z"
      >
        <header
          class="MuiBox-root css-1utcncn"
        >
          <div
            class="MuiBox-root css-axw7ok"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-zg625t-MuiTypography-root"
            >
              Eco
            </h4>
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/src/assets/logo.svg"
              style="color: transparent;"
              width="40"
            />
          </div>
        </header>
        <main
          class="MuiBox-root css-16too2"
        >
          <div>
            Test Content
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-21eb5z"
    >
      <header
        class="MuiBox-root css-1utcncn"
      >
        <div
          class="MuiBox-root css-axw7ok"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-zg625t-MuiTypography-root"
          >
            Eco
          </h4>
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/src/assets/logo.svg"
            style="color: transparent;"
            width="40"
          />
        </div>
      </header>
      <main
        class="MuiBox-root css-16too2"
      >
        <div>
          Test Content
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
